pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarStats.java

```diff
@@ -139,6 +139,8 @@ public synchronized void updateStats(
                                 // this task: helps to activate inactive-backlog-cursors which have caught up and
                                 // connected, also deactivate active-backlog-cursors which has backlog
                                 topic.checkBackloggedCursors();
+                                // check if topic is inactive and require ledger rollover
+                                ((PersistentTopic) topic).checkInactiveLedgers();
```

Comment:
Nit: I can see that `topic.checkBackloaggedCursors()` here too, but I wonder if this implementation tightly couples two distinct features. I doubt that someone would turn off the stats feature, but if they did, they'd lose this feature.

We already have an event loop used to delete inactive topics. The broker service initializes several inactivity related loops here: 

https://github.com/apache/pulsar/blob/f97d6e5a50355e1de628569c6de68cd7210094f0/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L522-L545

What is your perspective on adding this check to that loop? If we don't move the loop, I think we should document somewhere that the frequency of this method is governed by `statsUpdateFrequencyInSecs`.

Reply:
no, it doesn't depend on any feature. this background thread is being used for multiple simple topic checks such as backlogged-cursor or inactive-ledger checks. and it's internal implementation and we don't have to document it explicitly because it's just a safeguard check at the broker.

Reply:
It's fine if we want to leave it as is, but as it stands, the frequency of this method is determined by the broker config for `statsUpdateFrequencyInSecs`. In my view, that is unexpected coupling.

Reply:
as I explained earlier, this background thread performs topic's check including stats, backlogged-cursor-check and this will be an additional one. stats is something an essential part and its frequency should be reasonable enough to handle this kind of check. I don't see any usecase for why anyone disables the stats. however, if it's a real concern then will probably create another PR to separate all topic's checks out of this dedicated thread. but that can be addressed separately and definitely can not be a blocker for this PR.